src/main/scala/coreplex/Coreplex.scala

```diff
@@ -65,21 +80,82 @@ abstract class Coreplex(implicit val p: Parameters, implicit val c: CoreplexConf
 }
 
 class DefaultCoreplex(tp: Parameters, tc: CoreplexConfig) extends Coreplex()(tp, tc) {
+  class TileWrapper(tileBuilder: (Bool, Parameters) => Tile, _clock: Clock, _reset: Bool)
+      (implicit p: Parameters) extends Module(Some(_clock), Some(_reset)) {
+
+    val wrappedTile = tileBuilder(reset, p)
+    val io = new Bundle {
+      val tile = wrappedTile.io.cloneType
+      val ext_clock = Clock(INPUT)
+      val ext_reset = Bool(INPUT)
+    }
+
+    io.tile.uncached <> wrappedTile.io.uncached.map(AsyncClientUncachedTileLinkTo(io.ext_clock, io.ext_reset, _))
+    io.tile.cached <> wrappedTile.io.cached.map(AsyncClientTileLinkTo(io.ext_clock, io.ext_reset, _))
+    wrappedTile.io.interrupts := LevelFrom(io.ext_clock, io.tile.interrupts)
```

Comment:
I've taken a look at the PLIC and PRCI to see if synchronization can cause an unintended interrupt trigger.

For the PLIC at least, I'm satisfied that it most likely won't happen. The sequence is

* gateway gets interrupt, sets valid
* PLIC sees valid, sets ready low, sets interrupt high
* Tile sees interrupt, disables interrupts, starts running ISR
* ISR claims interrupt (through TL read)
* PLIC sets interrupt low, sends claim response
* ISR services interrupt, causing device sending interrupt to set interrupt low
* ISR sends completion to PLIC, which forwards to gateway. Tile re-enables interrupts and continues normal execution

The gateway won't send another interrupt to the PLIC until it gets the interrupt completion, so that's definitely safe. The only issue might be if the PLIC interrupt going low after the claim takes longer to settle on the Tile's domain than it takes to send the claim response, process it, and send the interrupt completion. I think that's very unlikely, since the interrupt level will settle after three or four clock cycles (of the Tile's clock), whereas processing the interrupt would involve several instructions to execute on the CPU and a potentially very long round trip through MMIO to the device that triggered the interrupt.

For the PRCI, the sequence is.

* time counter exceeds the timecmp value set by the tile, interrupt triggers
* timer interrupt routine runs, increases the timecmp value, interrupt goes low (on PRCI side)
* timer interrupt routine exits, re-enables interrupts

Here, it is possible that the interrupt level won't go low until after the timer interrupt routine exits. This depends on how long the timer routine continues to run after bumping the timecmp value.